lib/screens/v2/candidate_generator/pre_fare.ex

```diff
@@ -53,21 +53,27 @@ defmodule Screens.V2.CandidateGenerator.PreFare do
   def candidate_instances(
         config,
         now \\ DateTime.utc_now(),
-        elevator_status_instances_fn \\ &Widgets.ElevatorClosures.elevator_status_instances/2,
+        subway_status_instance_fn \\ &Widgets.SubwayStatus.subway_status_instances/1,
+        reconstructed_alert_instances_fn \\ &Widgets.ReconstructedAlert.reconstructed_alert_instances/1,
+        elevator_status_instance_fn \\ &Widgets.ElevatorClosures.elevator_status_instances/2,
         evergreen_content_instances_fn \\ &Widgets.Evergreen.evergreen_content_instances/1
       ) do
     [
-      fn -> elevator_status_instances_fn.(config, now) end,
-      fn -> placeholder_instances() end,
+      fn -> header_instances(config, now) end,
+      fn -> subway_status_instance_fn.(config) end,
+      fn -> reconstructed_alert_instances_fn.(config) end,
+      fn -> elevator_status_instance_fn.(config, now) end,
       fn -> evergreen_content_instances_fn.(config) end,
-      fn -> header_instances(config, now) end
+      fn -> placeholder_instances() end
     ]
     |> Task.async_stream(& &1.(), ordered: false, timeout: :infinity)
     |> Enum.flat_map(fn {:ok, instances} -> instances end)
   end
 
   defp header_instances(config, now) do
-    %Screen{app_params: %PreFare{header: %CurrentStopName{stop_name: stop_name}}} = config
+    %Screen{app_params: %PreFare{header: %CurrentStopId{stop_id: stop_id}}} = config
+
+    stop_name = Screens.V2.CandidateGenerator.BusShelter.fetch_stop_name(stop_id)
```

Comment:
I think it makes sense to fetch the name, but we should probably go ahead and move the fetch function to a utility file. I think `stop.ex` would be a good place for it that already exists.

Reply:
Ahh, I see the comments where you note this. So I'll let you decide between now or later.